rasa/shared/nlu/training_data/formats/dialogflow.py

```diff
@@ -102,24 +100,44 @@ def _flatten(list_of_lists: List[List[Any]]) -> List[Any]:
 
     @staticmethod
     def _extract_lookup_tables(
-        name: Text, examples: List[Dict[Text, Any]]
+        entity_js: Dict[Text, Any], examples_js: List[Dict[Text, Any]]
     ) -> Optional[List[Dict[Text, Any]]]:
-        """Extract the lookup table from the entity synonyms"""
-        synonyms = [e["synonyms"] for e in examples if "synonyms" in e]
+        """Extract the lookup table from the entity synonyms."""
+        synonyms = [e["synonyms"] for e in examples_js if "synonyms" in e]
         synonyms = DialogflowReader._flatten(synonyms)
         elements = [synonym for synonym in synonyms if "@" not in synonym]
 
         if len(elements) == 0:
             return None
-        return [{"name": name, "elements": elements}]
+        return [{"name": entity_js.get("name"), "elements": elements}]
+
+    @staticmethod
+    def _extract_regex_features(
+        entity_js: Dict[Text, Any], examples_js: List[Dict[Text, Any]]
+    ) -> Optional[List[Dict[Text, Any]]]:
+        """Extract the regex features from the entity synonyms."""
+        synonyms = [e["synonyms"] for e in examples_js if "synonyms" in e]
+        synonyms = DialogflowReader._flatten(synonyms)
+        return [
+            {"name": entity_js.get("name"), "pattern": synonym} for synonym in synonyms
+        ]
 
     @staticmethod
-    def _read_entities(entity_js, examples_js) -> "TrainingData":
+    def _read_entities(
+        entity_js: Dict[Text, Any], examples_js: List[Dict[Text, Any]]
+    ) -> "TrainingData":
         entity_synonyms = transform_entity_synonyms(examples_js)
 
-        name = entity_js.get("name")
-        lookup_tables = DialogflowReader._extract_lookup_tables(name, examples_js)
-        return TrainingData([], entity_synonyms, [], lookup_tables)
+        if entity_js["isRegexp"]:
+            regex_features = DialogflowReader._extract_regex_features(
+                entity_js, examples_js
+            )
+            return TrainingData([], entity_synonyms, regex_features, [],)
+        else:
+            lookup_tables = DialogflowReader._extract_lookup_tables(
+                entity_js, examples_js
+            )
+            return TrainingData([], entity_synonyms, [], lookup_tables,)
 
     @staticmethod
     def _read_examples_js(fn: Text, language: Text, fformat: Text) -> Any:
```

Comment:
I don't think it makes much sense.
In addition, `reads` method reads `TrainingData` from a string which is a weird thing to do in this case because DialogFlow project consists of a bunch of files.